src/state/translations-mixin.ts

```diff
@@ -58,18 +69,35 @@ export default <T extends Constructor<HassBaseEl>>(superClass: T) =>
       }
       const oldHass = changedProps.get("hass");
       if (this.hass?.panels && oldHass.panels !== this.hass.panels) {
-        this._loadFragmentTranslations(this.hass.language, this.hass.panelUrl);
+        this._loadFragmentTranslations(
+          this.hass.locale?.language,
+          this.hass.panelUrl
+        );
       }
     }
 
-    protected hassConnected() {
+    protected async hassConnected() {
       super.hassConnected();
-      getUserLanguage(this.hass!).then((language) => {
-        if (language && this.hass!.language !== language) {
-          // We just get language from backend, no need to save back
-          this._selectLanguage(language, false);
-        }
-      });
+      const result = await fetchTranslationPreferences(this.hass!);
+      const userNumberFormat = result?.number_format;
+      const userLanguage = result?.language
+        ? findAvailableLanguage(result.language)
+        : null;
+
+      const prefs: FrontendTranslationData = {
+        language: userLanguage || this.hass!.locale?.language,
+        number_format: userNumberFormat || this.hass!.locale?.number_format,
+      };
+
+      if (
+        result &&
+        (this.hass!.locale?.number_format !== userNumberFormat ||
+          this.hass!.locale?.language !== userLanguage)
+      ) {
+        // We just get language and number_format from backend, no need to save back
+        this._selectLanguage(prefs, false);
```

Comment:
We should not call `_selectLanguage` when just the `number_format` is changed, as that will reload all translations etc. and that is not needed ofc. Just when `language` is changed we should call `_selectLanguage`, if it is just `number_format` call `this._updateHass`.

Reply:
Good point. I am now firing a separate event for selecting the `number_format` and a separate event listener that just updates `hass.locale`.